### ISSUE TITLE
ALL variables fail

### ISSUE THREAD

**Issue by Pwntus** [serverless/serverless#3606]
# This is a Bug Report

## Description

Updated serverless 1.11.0 -> 1.13.0. Everything was fine on 1.11.0.

I first discovered this while deploying a stack which had a Cognito user pool setup as an authorizer for a API Gateway endpoint.

When deploying to AWS I get this:

```
Serverless Error ---------------------------------------
Template format error: Resource name UserUnderscorepool}ApiGatewayAuthorizer
is non alphanumeric.
```
`UserUnderscorepool}ApiGatewayAuthorizer` doesn't sound right.

When removing the affected function, I'll get something similar to this:

```
  Serverless Error ---------------------------------------
 
     An error occurred while provisioning your stack: VideoLogGroup
     - 1 validation error detected: Value '/aws/lambda/${self:provider.stage}-video'
     at 'logGroupNamePrefix' failed to satisfy constraint:
     Member must satisfy regular expression pattern: [\.\-_/#A-Za-z0-9]+.
```
When I then take a look at the `.serverless/cloudformation-template-update-stack.json` I can see all my variables embedded in the JSON object. E.g:

```JSON
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Resources": {
    "ServerlessDeploymentBucket": {
      "Type": "AWS::S3::Bucket"
    },
    "VideoLogGroup": {
      "Type": "AWS::Logs::LogGroup",
      "Properties": {
        "LogGroupName": "/aws/lambda/${self:provider.stage}-video"
      }
    }
}
```

I'm not sure, but shouldn't the variables have been replaced at this point?

Similar or dependent issues:
* #3511 

## Additional Data

* ***Serverless Framework Version you're using***: 1.13.0
* ***Operating System***: Ubuntu 17.04

**Comment by eahefnawy** [comment]
@Pwntus thanks for reporting. Duplicate of #3587

A patch will be released within 10 mins... 😊

**Comment by eahefnawy** [comment]
If you're curious on what went wrong, here's the fix => #3604 🙌

**Comment by eahefnawy** [comment]
@Pwntus I just released the patch 🎉 (v1.13.1) ... would be great if you update and confirm that everything is now working fine 😊
